lib/async_hooks.js

```diff
@@ -516,3 +517,10 @@ Object.defineProperty(module.exports, 'triggerId', {
   }, 'async_hooks.triggerId is deprecated. ' +
      'Use async_hooks.triggerAsyncId instead.', 'DEP0071')
 });
+
+// Integration with V8 Inspector AsyncTask* API
+// FIXME(bajtos) move this code to src/inspector_agent.cc
+// Agent::Start() should enable the hook, Agent::Stop() should disable it
```

Comment:
@bajtos 
> Typically `Agent::Start` is called directly from `node::Start`, long time before `bootstrap_node.js` is loaded. When debugging via `--inspect-brk`, `bootstrap_node.js` is not started until a debugger is attached.

To rephrase the issue, you can't create a hook until after `bootstrap_node.js` has run, and this is an issue with `--inspect-brk` because the hook won't be in place in time.

> I find the user experience of this solution rather poor. Because async stack trace tracing is enabled only at the time the debugger connects, any async transitions created at startup time (e.g. like setting up an HTTP server, creating timeouts, etc.) are lost.

If you were to add a `setupInspectorHook()` as the first call to `startup()` in `lib/internal/bootstrap_node.js` it would force the hook to be enabled before any possible asynchronous tasks are performed.

> At the same time, in order to support `process._startDebug`, I'll still need to pass the "enable" function into C++ land, so that we can enable the hook at runtime when the debugger was enabled via a signal (`_startDebug`).

To double check, this wouldn't start capturing async stack traces until after the process has been signaled, but on the same note does Chrome always record all async stack traces so they are always available after the inspector connects?